File: eKnjigaDesktopUI/lib/models/userReport.dart
class UserReport {
  final int id;
  final String reason;
  final int status;
  final DateTime createdAt;
  final User userReported;
  final User reportedByUser;

  UserReport({
    required this.id,
    required this.reason,
    required this.status,
    required this.createdAt,
    required this.userReported,
    required this.reportedByUser,
  });

  factory UserReport.fromJson(Map<String, dynamic> json) {
    return UserReport(
      id: json['id'],
      reason: json['reason'],
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
      userReported: User.fromJson(json['userReported']),
      reportedByUser: User.fromJson(json['reportedByUser']),
    );
  }
}

class User {
  final int id;
  final String firstName;
  final String lastName;

  User({
    required this.id,
    required this.firstName,
    required this.lastName,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
    );
  }
}

File: eKnjiga/eKnjiga.Services/Database/UserReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eKnjiga.Model.Enums;

namespace eKnjiga.Services.Database
{
    public class UserReport
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Reason { get; set; } = string.Empty;

        [Required]
        public UserReportStatus Status { get; set; } = UserReportStatus.Pending; 
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? ProcessedAt { get; set; }

        // Foreign key for reported User
        public int UserReportedId { get; set; }
        [ForeignKey("UserReportedId")]
        public User UserReported { get; set; }

        // Foreign key for User who reported
        public int ReportedByUserId { get; set; }
        [ForeignKey("ReportedByUserId")]
        public User ReportedByUser { get; set; }

        // Foreign key for User who processed the report
        public int? ProcessedByUserId { get; set; }
        [ForeignKey("ProcessedByUserId")]
        public User? ProcessedByUser { get; set; }
    }
} 